### PR TITLE
Abstracts database storage & Status page UI

### PR DESCRIPTION
**Pull request by gosom**
Added an IStore interface and an implementation of the current json storage.
This will help adding support for other databases like sqlite

Added also a Status page which displays all the peers

sample of how it looks like here:

https://prnt.sc/1p5oagi

### CODE REVIEW

File: templates/status.html [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            </div>
+        </div>
+        <!-- /.modal-content -->
+    </div>
+    <!-- /.modal-dialog -->
+</div>
+<!-- /.modal -->
+{{end}}
+
+{{define "bottom_js"}}
```
Comment (ngoduykhanh) [inline templates/status.html:85]: It looks like you cloned from the `global_settings.html` template. Do we need the following bottom_js scripts here?

File: templates/status.html [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+              {{ end }}
+              </tbody>
+            </table>
+        {{ end }}
+
+
+    </div>
+</section>
+
+<div class="modal fade" id="modal_endpoint_address_suggestion">
```
Comment (ngoduykhanh) [inline templates/status.html:58]: And this modal as well.

File: docker-compose.yaml [hunk 3]
```diff
@@ -5,13 +5,14 @@ services:
...
+    cap_add:
+      - NET_ADMIN
+    network_mode: host
     environment:
       - SENDGRID_API_KEY
       - EMAIL_FROM
       - EMAIL_FROM_NAME
       - SESSION_SECRET
-    ports:
-      - 5000:5000
```
Comment (ngoduykhanh) [inline docker-compose.yaml:14]: Why don't we want to expose the port here?

**Review by ngoduykhanh — CHANGES_REQUESTED** [review]

**Comment by ngoduykhanh** [comment]
Thanks @gosom for the PR. I've left a few comments there :-)

File: docker-compose.yaml (hunk 3 above)
Comment (gosom) [reply docker-compose.yaml:17]: Hi @ngoduykhanh , thanks for the review. 

In order to get the status from the `wg` command the docker-compose runs with network-mode host. 
This means that we do not need to expose any port

File: templates/status.html (hunk 1 above)
Comment (gosom) [reply templates/status.html]: removed

**Review by ngoduykhanh — APPROVED** [review]

**Pull request merged**